analyzers/src/SonarAnalyzer.Common/Helpers/WildcardPatternMatcher.cs

```diff
@@ -0,0 +1,134 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Concurrent;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Helpers;
+
+internal static class WildcardPatternMatcher
+{
+    public static bool IsMatch(string pattern, string input) =>
+        !(string.IsNullOrWhiteSpace(pattern) || string.IsNullOrWhiteSpace(input))
+        && WildcardPattern.Create(pattern).Match(input);
+
+    /// <summary>
+    /// Copied from https://github.com/SonarSource/sonar-plugin-api/blob/a9bd7ff48f0f77811ed909070030678c443c975a/sonar-plugin-api/src/main/java/org/sonar/api/utils/WildcardPattern.java.
+    /// </summary>
+    private sealed class WildcardPattern
+    {
+        private const string SpecialChars = "()[]^$.{}+|";
+        private static readonly ConcurrentDictionary<string, WildcardPattern> Cache = new();
+        private readonly Regex pattern;
+
+        private WildcardPattern(string pattern, string directorySeparator) =>
+            this.pattern = new Regex(ToRegexp(pattern, directorySeparator), RegexOptions.Compiled, RegexConstants.DefaultTimeout);
+
+        public bool Match(string value)
+        {
+            value = value.TrimStart('/');
+            value = value.TrimEnd('/');
+            try
+            {
+                return pattern.IsMatch(value);
+            }
+            catch (RegexMatchTimeoutException)
+            {
+                return false;
+            }
+        }
+
+        public static WildcardPattern Create(string pattern) =>
+            Create(pattern, Path.DirectorySeparatorChar.ToString());
+
+        private static WildcardPattern Create(string pattern, string directorySeparator) =>
+            Cache.GetOrAdd(pattern + directorySeparator, _ => new WildcardPattern(pattern, directorySeparator));
+
+        private static string ToRegexp(string wildcardPattern, string directorySeparator)
+        {
+            var escapedDirectorySeparator = '\\' + directorySeparator;
+            var sb = new StringBuilder(wildcardPattern.Length);
+
+            sb.Append('^');
+
+            var i = wildcardPattern.StartsWith("/") || wildcardPattern.StartsWith("\\") ? 1 : 0;
+            while (i < wildcardPattern.Length)
+            {
+                var ch = wildcardPattern[i];
+
+                if (SpecialChars.IndexOf(ch) != -1)
+                {
+                    // Escape regexp-specific characters
+                    sb.Append('\\').Append(ch);
+                }
```

Comment:
This looks like java-specific implementation of something that is done by `Regex.Escape`

Reply:
Fixed by #6919